sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1096 @@
+    private String[] packages = new String[0];
+    /** Parameter: packages to exclude from checking. */
+    private String[] excludePackages = new String[0];
+    /** Parameter: overriding return value annotations allowed. */
+    private boolean allowOverridingReturnValue;
+    /** Parameter: overriding parameter annotations allowed. */
+    private boolean allowOverridingParameter;
+
+    /** State, is a package excluded. */
+    private boolean packageExcluded;
+
+    @Override
+    public final int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public final int[] getRequiredTokens() {
+        return new int[0];
+    }
+
+    @Override
+    public final int[] getAcceptableTokens() {
+        return DEFAULT_MODIFIERS.clone();
+    }
+
+    @Override
+    public final void visitToken(final DetailAST ast) {
+        if (ast.getType() == TokenTypes.PACKAGE_DEF) {
+            final DetailAST nameAST = ast.getLastChild().getPreviousSibling();
+            packageExcluded = isPackageExcluded(FullIdent.createFullIdent(nameAST));
+        }
+        else if (!packageExcluded) {
+            final AbstractJsr305Handler handler = handleDefinition(ast);
+            if (handler != null) {
+                handler.check();
+            }
+        }
+    }
+
+    /**
+     * Sets the included packages parameter.
+     * @param packageNames
+     *        the package names, comma separated
+     */
+    public void setPackages(final String... packageNames) {
+        packages = transformToUnique(packageNames);
+    }
+
+    /**
+     * Maps annotations to their respective names.
+     * @return the map
+     */
+    private static Map<String, NullnessAnnotation> createString2AnnotationMap() {
+        final Map<String, NullnessAnnotation> result = new HashMap<>();
+
+        for (final NullnessAnnotation annotation : NullnessAnnotation.values()) {
+            result.put(annotation.annotationName, annotation);
+            result.put(annotation.fullyQualifiedClassName, annotation);
+        }
+
+        return Collections.unmodifiableMap(result);
+    }
+
+    /**
+     * Sets the excluded packages parameter.
+     * @param packageNames
+     *        the package names, comma separated
+     */
+    public void setExcludePackages(final String... packageNames) {
+        excludePackages = transformToUnique(packageNames);
+    }
+
+    /**
+     * Removes duplicates from an array of strings.
+     * @param input
+     *        the array
+     * @return a new, duplicate-free array
+     */
+    private static String[] transformToUnique(final String... input) {
+        final Set<String> inputSet = new HashSet<>(Arrays.asList(input));
+        return inputSet.toArray(new String[0]);
+    }
+
+    /**
+     * Checks whether a package is excluded.
+     * @param fullIdent
+     *        the identifier
+     * @return true if yes
+     */
+    protected boolean isPackageExcluded(final FullIdent fullIdent) {
```

Comment:
all methods should be private in custom Check, only overridden methods are public.

Reply:
Done.

Reply:
Strange, looks like you are still looking at an outdated version of the code. I squashed my commits and pushed them on May 16, and in the [latest code](https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/742/commits/04cb7fa1767455853d0c5ad6583dfba4e0e3536a#diff-424509c2f1bc401479715d7d8a467cef) it looks just fine.